File: leetcode/0121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0:
            return 0
        curr_profit, curr_enter = 0, prices[0]
        for i in range(1, len(prices)):
            if prices[i] < curr_enter:
                curr_enter = prices[i]
            else:
                curr_profit = max(curr_profit, prices[i] - curr_enter)
        return curr_profit

File: leetcode/0121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy = prices[0], max_profit = 0;
        for(int i = 1; i < prices.size(); ++i)
        {
            if(prices[i] < buy)
                buy = prices[i];
            else
                max_profit = max(max_profit, prices[i] - buy);
        }
        return max_profit;
    }
};
